qualtran/bloqs/arithmetic/multiplication.py

```diff
@@ -109,13 +110,23 @@ class Square(Bloq):
     """
 
     bitsize: int
+    uncompute: bool = False
```

Comment:
is this just to support `on_classical_vals`? the rest of the adjointedness is formulaic, right? Can you note these things in an issue. I'd like to prevent the re-introduction of `is_adjoint` flags on every bloq. @tanujkhattar

Reply:
It's also to support the `side` of the register. But maybe that's moot now and `Adjoint(Square(x))` would take care of it automatically?

Reply:
yes, side switching is part of `Adjoint`. xref #660